File: solutions/python/bob/1/bob.py
def response(hey_bob):
    stripped = hey_bob.strip()
    
    if not stripped:
        return "Fine. Be that way!"
    
    is_yelling = stripped.upper() == stripped and any(c.isalpha() for c in stripped)
    is_question = stripped.endswith("?")
    
    if is_yelling and is_question:
        return "Calm down, I know what I'm doing!"
    elif is_yelling:
        return "Whoa, chill out!"
    elif is_question:
        return "Sure."
    else:
        return "Whatever."

File: solutions/javascript/bob/1/bob.js
export const hey = (message) => {
  const trimmed = message.trim();

  const isSilence = trimmed.length === 0;
  const isYelling = /[A-Z]/.test(trimmed) && trimmed === trimmed.toUpperCase();
  const isQuestion = trimmed.endsWith('?');

  if (isSilence) return 'Fine. Be that way!';
  if (isYelling && isQuestion) return "Calm down, I know what I'm doing!";
  if (isYelling) return 'Whoa, chill out!';
  if (isQuestion) return 'Sure.';
  return 'Whatever.';
};